scalyr_agent/monitor_utils/k8s.py

```diff
@@ -971,7 +994,13 @@ def update_cache( self, run_state ):
             except Exception, e:
                 global_log.warn( "Exception occurred when updating k8s cache - %s\n%s" % (str( e ), traceback.format_exc()) )
 
-            run_state.sleep_but_awaken_if_stopped( local_state.cache_expiry_secs )
+            # Fuzz how much time we spend until the next cycle.  This should spread out when the agents query the
+            # API master over time in clusters with a larger number of agents.
+            if local_state.cache_expiry_fuzz_secs > 0:
+                fuzz_factor = random.uniform(0, local_state.cache_expiry_fuzz_secs)
+            else:
+                fuzz_factor = 0
+            run_state.sleep_but_awaken_if_stopped( local_state.cache_expiry_secs - fuzz_factor )
```

Comment:
What happens if `fuzz_factor` is greater than `cache_expiry_secs`?

Reply:
Yup.  I changed this to do a max with zero so we never try to sleep a negative amount.  We might be sleeping zero.. but .. they asked for it.